packages/core/src/external/canvas/note.ts

```diff
@@ -0,0 +1,85 @@
+import CanvasSDK from "./index";
+import { generateFakeBundleFemale, generateFakeBundleMale } from "./data";
+import { out } from "../../util/log";
+const { log } = out("[CANVAS NOTE]");
+
+export async function createFullNote({
+  canvas,
+  canvasPatientId,
+  patientA,
+  patientB,
+}: {
+  canvas: CanvasSDK;
+  canvasPatientId: string;
+  patientA: boolean;
+  patientB: boolean;
+}) {
+  try {
+    const canvasPractitioner = await canvas.getPractitioner("Wilson");
+    const canvasPractitionerId = canvasPractitioner.id;
+
+    const canvasLocation = await canvas.getLocation();
+    const canvasLocationId = canvasLocation.id;
+
+    const canvasEncounter = await canvas.getFirstEncounter(canvasPatientId);
+    const canvasEncounterId = canvasEncounter.id;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const canvasNoteId = (canvasEncounter.extension as any)[0].valueId;
+
+    if (!canvasLocationId || !canvasPractitionerId || !canvasEncounterId) {
+      throw new Error("Canvas Location ID or Practitioner ID is undefined");
+    }
+
+    console.log(`[CANVAS-EVENT-LISTENER] Creating canvas resources for patient ${canvasPatientId}`);
+    let data;
+    if (patientA) {
+      data = generateFakeBundleFemale(canvasPatientId, canvasPractitionerId, canvasEncounterId);
+    } else if (patientB) {
+      data = generateFakeBundleMale(canvasPatientId, canvasPractitionerId, canvasEncounterId);
+    } else {
+      throw new Error("Either patientA or patientB must be true");
+    }
+
+    if (!data.entry) {
+      throw new Error("Consolidated patient data is undefined");
+    }
+
+    for (const entry of data.entry) {
+      const resource = entry.resource;
+      if (!resource) {
+        throw new Error("Resource is undefined");
```

Comment:
Could use `MetriportError` so we can add extra into on the third parameter of the constructor, like the parameters for createFullNote that led to this error